keras/applications/convnext.py

```diff
@@ -0,0 +1,661 @@
+    config.update({"drop_path_rate": self.drop_path_rate})
+    return config
+
+
+class ConvNeXtBlock(Model):
+  """ConvNeXt block.
+  
+  References:
+    - https://arxiv.org/abs/2201.03545
+    - https://github.com/facebookresearch/ConvNeXt/blob/main/models/convnext.py
+
+  Notes:
+    In the original ConvNeXt implementation (linked above), the authors use
+    `Dense` layers for pointwise convolutions for increased efficiency. 
+    Following that, this implementation also uses the same.
+
+  Args:
+    projection_dim (int): Number of filters for convolution layers. In the
+    ConvNeXt paper, this is referred to as projection dimension.
+    drop_path (float): Probability of dropping paths. Should be within [0, 1].
+    layer_scale_init_value (float): Layer scale value. Should be a small float
+      number.
+
+  Returns:
+    A keras.Model instance.
+  """
+  def __init__(self, projection_dim, drop_path_rate=0.0, 
+    layer_scale_init_value=1e-6, **kwargs):
+    super().__init__(**kwargs)
+    self.projection_dim = projection_dim
+    self.drop_path_rate = drop_path_rate
+    self.layer_scale_init_value = layer_scale_init_value
+    name = kwargs["name"]
+    
+    if layer_scale_init_value > 0.0:
+      self.gamma = tf.Variable(
+        layer_scale_init_value * tf.ones((projection_dim,)),
+        name=name + "_layer_scale_gamma")
+    else:
+      self.gamma = None
+    
+    self.dw_conv_1 = layers.Conv2D(
+      filters=projection_dim, kernel_size=7, padding="same",
+      groups=projection_dim, name=name + "_depthwise_conv")
+    self.layer_norm = layers.LayerNormalization(epsilon=1e-6, 
+      name=name + "_layernorm")
+    self.pw_conv_1 = layers.Dense(4 * projection_dim,
+      name=name + "_pointwise_conv_1")
+    self.act_fn = layers.Activation("gelu", name=name + "_gelu")
+    self.pw_conv_2 = layers.Dense(projection_dim, 
+      name=name + "_pointwise_conv_2")
+    self.drop_path = (
+      StochasticDepth(drop_path_rate, name=name + "_stochastic_depth")
+      if drop_path_rate > 0.0
+      else layers.Activation("linear", name=name + "_identity")
+    )
+
+  def call(self, inputs):
+    x = inputs
+
+    x = self.dw_conv_1(x)
+    x = self.layer_norm(x)
+    x = self.pw_conv_1(x)
+    x = self.act_fn(x)
+    x = self.pw_conv_2(x)
+
+    if self.gamma is not None:
+      x = self.gamma * x
+
+    return inputs + self.drop_path(x)
+
+  def get_config(self):
+    config = {
+      "projection_dim": self.projection_dim,
+      "drop_path_rate": self.drop_path_rate,
+      "layer_scale_init_value": self.layer_scale_init_value,
+    }
+    return config
+
+
+def PreStem(name=None):
+  """Normalizes inputs with ImageNet-1k mean and std.
+
+  Args:
+    name (str): Name prefix.
+
+  Returns:
+    Normalized tensor.
+  """
+  if name is None:
+    name = "prestem" + str(backend.get_uid("prestem"))
+
+  def apply(x):
+    x = layers.Normalization(
+      mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
```

Comment:
So, these are initialized based on imagenet: this is required for use with the pretrained weights.  Is there a way we can allow users to configure this for custom datasets?

@fchollet

Reply:
It's inspired from ResNet-RS and RegNets. For non-ImageNet weights, it's necessary to disable it.